pkg/obfuscator/tracker.go

```diff
@@ -106,6 +106,15 @@ func (s *SimpleTracker) Initialize(report ReplacementReport) {
 	}
 }
 
-func NewSimpleTracker() ReplacementTracker {
-	return &SimpleTracker{mapping: map[string]*Replacement{}}
+func NewSimpleTracker(existingReplacements map[string]string) ReplacementTracker {
+	var tracker = &SimpleTracker{mapping: map[string]*Replacement{}}
+	tracker.lock.Lock()
```

Comment:
if you reverse the order of the statements, you can avoid the locking:

```
var m map[string]*Replacement{}        
for key, value := range existingReplacements {
   m[key] = NewReplacement(key, key, value, 0)
}
return &SimpleTracker{mapping: m}
```

Reply:
also to mention: there is no way any other goroutine could've changed your tracker here, so that lock is also not necessary in the first place